libopenage/gamestate/event/spawn_entity.cpp

```diff
@@ -71,28 +71,28 @@ void SpawnEntityHandler::invoke(openage::event::EventLoop & /* loop */,
 
 	auto pos = params.get("position", gamestate::WORLD_ORIGIN);
 	entity_pos->set_position(time, pos);
-
-	// TODO: Testing values
-	entity_pos->set_position(time + 1, pos + coord::phys3_delta{0, 1, 0});
-	entity_pos->set_position(time + 2, pos + coord::phys3_delta{1, 2, 0});
-	entity_pos->set_position(time + 3, pos + coord::phys3_delta{2, 2, 0});
-	entity_pos->set_position(time + 4, pos + coord::phys3_delta{3, 1, 0});
-	entity_pos->set_position(time + 5, pos + coord::phys3_delta{3, 0, 0});
-	entity_pos->set_position(time + 6, pos + coord::phys3_delta{2, -1, 0});
-	entity_pos->set_position(time + 7, pos + coord::phys3_delta{1, -1, 0});
-	entity_pos->set_position(time + 8, pos);
-
 	entity_pos->set_angle(time, coord::phys_angle_t::from_int(315));
-	entity_pos->set_angle(time + 1, coord::phys_angle_t::from_int(279));
-	entity_pos->set_angle(time + 2, coord::phys_angle_t::from_int(225));
-	entity_pos->set_angle(time + 3, coord::phys_angle_t::from_int(180));
-	entity_pos->set_angle(time + 4, coord::phys_angle_t::from_int(135));
-	entity_pos->set_angle(time + 5, coord::phys_angle_t::from_int(90));
-	entity_pos->set_angle(time + 6, coord::phys_angle_t::from_int(45));
-	entity_pos->set_angle(time + 7, coord::phys_angle_t::from_int(0));
-	entity_pos->set_angle(time + 8, coord::phys_angle_t::from_int(315));
-
-	auto test_pos = std::make_shared<component::Position>(this->loop, pos, time);
+
+	if (entity->has_component(component::component_t::MOVE)) {
```

Comment:
What do you think about this code? 
```
if (entity->has_component(component::component_t::MOVE)) {
    std::vector<coord::phys3_delta> positions = {
        {0, 1, 0},
        {1, 2, 0},
        {2, 2, 0},
        {3, 1, 0},
        {3, 0, 0},
        {2, -1, 0},
        {1, -1, 0},
        {0, 0, 0} // Assuming this is equivalent to pos
    };

    std::vector<int> angles = {
        279,
        225,
        180,
        135,
        90,
        45,
        0,
        315
    };

    for (int i = 0; i < positions.size(); i++) {
        entity_pos->set_position(time + i + 1, pos + positions[i]);
        entity_pos->set_angle(time + i + 1, coord::phys_angle_t::from_int(angles[i]));
    }
}
```

Reply:
Oh hey, thanks for checking out the code :D

Given that the current code is just test code and soon to be removed, it probably doesn't really matter how it looks. Your solution looks more maintainable though, so if this were an engine demo, it would be the better solution.